acquire/outputs/tar.py

```diff
@@ -26,23 +42,51 @@ def __init__(self, path, compress=False, encrypt=False, public_key=None, **kwarg
         else:
             self.tar = tarfile.open(name=self.path, mode=mode)
 
-    def write(self, path, fh, size=None, entry=None):
+    def write(
+        self,
+        path: Path,
+        fh: BinaryIO,
+        size: Optional[int] = None,
+        entry: Optional[FilesystemEntry] = None,
+    ) -> None:
+        """Write a filesystem entry or file-like object to a tar file.
+
+        Args:
+            path: The path of the entry to write.
+            fh: The file-like object of the entry to write.
+            size: The optional file size in bytes of the entry to write.
+            entry: the optional filesystem entry of the entry to write.
```

Comment:
```suggestion
            entry: The optional filesystem entry of the entry to write.
```

Reply:
Also noting that this might still be a wrong type hint, see other comment.

Reply:
I believe the `path` variable here primarily gets used to constructs (arbitrary) paths to map entires into their respective output format, like `$rootfs$/etc/hostname` to `/etc/hostname`. I'am thinking about changing `path` to `output_path` and changing the docstring as well to better reflect this.

I will fix the `Path` and `FilesystemEntry` mismatch  in the process as well. Of the other functions as well.